src/components/SearchOverlay/SearchOverlay.js

```diff
@@ -104,6 +105,22 @@ class SearchOverlay extends React.PureComponent {
     }
   }
 
+  componentDidMount() {
+    core.addEventListener('searchFailed', this.searchFailed);
```

Comment:
Do you think we can listen to this event in `SearchPanel`? If we do this then we don't need to have an extra state and action for `setSearchError`. We can instead having an internal error state in `SearchPanel`.

Reply:
For this one, I think I just followed how `isSearching` and `noResult` are set in `SearchOverlay` and it flows down to SearchPanel. I thought in the way we keep `SearchPanel` simpler component.

If you think it should be in `SearchPanel` in its own state, let me know I can change it.